common/memory.h

```diff
@@ -27,6 +27,14 @@
 
 namespace Common {
 
+/**
+ * @defgroup common_memory Memory
+ * @ingroup common
+ *
+ * @brief Functions for managing the memory.
```

Comment:
Same as earlier - we should have consistent terminology - functions, api, class, methods

Reply:
I guess in this case the most accurate description is 'Class templates for managing the memory' ?

Reply:
Referring to classes here seems confusing to me. They look like `Template functions`.